chainstate/src/detail/mod.rs

```diff
@@ -688,5 +796,56 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
     }
 }
 
+#[derive(Error, Debug, PartialEq, Eq, Clone)]
+enum BlockIntegrationError {
+    #[error("Reorg error during block integration: {0}; resulting block status is {1}; first bad block id is {2}")]
+    ConnectBlockErrorDuringReorg(BlockError, BlockStatus, Id<Block>),
+    #[error("Generic error during block integration: {0}; resulting block status is {1}")]
+    OtherValidationError(BlockError, BlockStatus),
+    #[error("Failed to commit block data for block {0} after {1} attempts: {2}")]
+    BlockCommitError(Id<Block>, usize, chainstate_storage::Error),
+    #[error("Database error: {0}")]
+    OtherDbError(#[from] chainstate_storage::Error),
+}
+
+fn block_integration_error_constructor<Err>(
+    status: BlockStatus,
+) -> impl FnOnce(Err) -> BlockIntegrationError
+where
+    Err: Into<BlockError>,
+{
+    move |err| BlockIntegrationError::OtherValidationError(err.into(), status)
+}
+
+fn get_block_index<S, V>(
+    chainstate_ref: &chainstateref::ChainstateRef<S, V>,
+    block_id: &Id<Block>,
+) -> Result<Option<BlockIndex>, BlockError>
+where
+    S: BlockchainStorageRead,
+    V: TransactionVerificationStrategy,
+{
+    chainstate_ref
+        .get_block_index(block_id)
+        .map_err(|err| BlockError::BlockIndexQueryError(err, (*block_id).into()))
+}
+
+fn get_existing_block_index<S, V>(
+    chainstate_ref: &chainstateref::ChainstateRef<S, V>,
+    block_id: &Id<Block>,
+) -> Result<BlockIndex, BlockError>
+where
+    S: BlockchainStorageRead,
+    V: TransactionVerificationStrategy,
+{
+    get_block_index(chainstate_ref, block_id)?.ok_or(BlockError::InvariantErrorBlockIndexNotFound(
+        (*block_id).into(),
+    ))
+}
+
+fn get_next_block_height(height: BlockHeight) -> Result<BlockHeight, BlockError> {
+    (height + 1.into()).ok_or(BlockError::BlockHeightTooBig(height))
```

Comment:
I believe we don't care in the rest of the code and just call `BlockHeight::next_height` method. So do we need this method?

Reply:
I've just missed the fact that BlockHeight::next_height exists.